### ISSUE TITLE
mpdata (from CoD4) is considered invalid

### ISSUE THREAD

**Issue by PureTryOut** [spmn/iwmtool#1]
Hi, thanks for this program! It's hard to find any useful tools like this for such an old game anymore.

I managed to compile iwmtool locally (on Linux, using a custom written `CMakeLists.txt`, I can PR it if you want) but it fails to decrypt my mpdata. I verified the key code and it's correct but the tool reports `Error: Input stats file "/path/to/mpdata" appears to be invalid, err: InvalidIWMHash` (obviously replacing the path with the actual path to the mpdata file).

Another tool, `codmpdatadec` found at http://old.zenhax.com/call-of-duty-mpdata-file-t824.html, managed to decrypt it but the resulting file is still a binary blob and without iwmtool I don't really know how I would edit it. Pointing iwmtool at that decrypted file gives the same result sadly.

**Comment by spmn** [comment]
Hi, I'm a bit surprised that there is (at least) one user trying out this tool 😄 

Sadly, I haven't touched CoD4 in a while, so my knowledge on the subject is going to be very rusty. However, I can take a look at it if you provide me with the faulty mpdata file and the associated CD-key. If you're ok with that, you can email me the key and the file (I assume you wouldn't be comfortable with pasting them here in plain view :) )

**Comment by PureTryOut** [comment]
You're right, I'm not comfortable with that lol. Even though being such an old game you can just get a random cd key from a Youtube video, I like to keep mine legit and private :wink: 

I've send you an email!

**Comment by spmn** [comment]
I will take a look at them this weekend

**Comment by spmn** [comment]
Pushed a fix to master. Try it out and let me know if that solved the Linux issue.

**Comment by spmn** [comment]
I think it should be fixed now. If not, reopen this issue.

**Comment by PureTryOut** [comment]
Works! Now I need to find of indices I can manipulate... Would be good if there was a wiki page somewhere :sweat_smile:

**Comment by spmn** [comment]
A few indices (they are internally called `statids`) are hardcoded in the `.exe` file, but most of them (especially the ones related to stats, challenges, perks, weapon unlocks etc.) can be extracted from the multiplayer `.ff` files.

Back in 2019 when I made this tool, I also wanted to make a list of all indices, but they are scattered around multiple game scripts, csv tables and config files, so I gave up.

Anyway, here is what I managed to dump from the game files when I was still optimistic about it 😃. Maybe they'll be useful to you. You'd have to focus on the files inside `.\code_post_gfx_mp\raw\mp\` and how they are referenced from other (usually gsc) files.

[cod4_english_dump_files.zip](https://github.com/spmn/iwmtool/files/9996128/cod4_english_dump_files.zip)
